bms-monitor.js

```diff
@@ -1,18 +1,110 @@
-const {expect} = require('chai');
+const { expect } = require('chai');
+const {
+	SocRange,
+	RangeValues,
+	ChargeRange,
+	TemperatureRange,
+} = require('./constants');
+const BatteryStatus = require('./properties-monitor');
+
+const tempMinValue = RangeValues.tempMinValue;
+const tempMaxValue = RangeValues.tempMaxValue;
+const socMinValue = RangeValues.socMinValue;
+const socMaxValue = RangeValues.socMaxValue;
+const chargeRateMinValue = RangeValues.chargeRateMinValue;
+const chargeRateMaxValue = RangeValues.chargeRateMaxValue;
+const TolerenceValue = RangeValues.tolerence;
+
+const batteryStatusCheck = (type, value) => {
+	var statusType = {
+		temperature: function () {

```

Comment:
getTolerance, checkWarning and checkMinMaxValue seems to be similar code repeating 3 times. Can this be organised in a better way? please check.